nibe/connection/nibegw.py

```diff
@@ -120,13 +122,23 @@ def datagram_received(self, data, addr):
                 e,
             )
 
+    async def read_product(self, timeout: float = READ_PRODUCT_TIMEOUT) -> Product:
+        self._futures["product"] = asyncio.get_event_loop().create_future()
+        try:
+            result = await asyncio.wait_for(self._futures["product"], timeout)
+            return Product(result["model"], result["version"])
+        except asyncio.TimeoutError:
+            raise CoilReadTimeoutException(f"Timeout waiting for product message")
```

Comment:
Not a coil read exception. Good to keep separate i think.

Reply:
Agree

Reply:
I am thinking to remove Coil from all Exception names.

ReadException, ReadTimeoutException, WriteException, WriteTimeoutException

Reply:
Well coil read is different from product. One is passive one is active. I think its good to separate.